docs/spec.md

```diff
@@ -136,6 +138,16 @@ QuantizationScale ::= FloatConstant
 QuantizationZeroPoint ::= IntegerConstant
 ```
 
+| Name                     | Type                                        | Constraints           |
+|--------------------------|---------------------------------------------|-----------------------|
+| `storage_type`           | integer type                                | (C1-C4), (C9)         |
+| `storage_min`            | integer constant                            | (C2), (C4), (C8)      |
+| `storage_max`            | integer constant                            | (C3), (C4), (C8)      |
+| `expressed_type`         | floating-point type                         | (C1), (C5)            |
+| `quantization_dimension` | optional integer constant                   | (C12), (C13)          |
```

Comment:
I think that `quantization_dimension` is genuinely optional, i.e. it doesn't have a reasonable default value unlike everything else in the spec, so I think there's no way out of this. (And I'll also add `is_empty` to supported functions).

In principle, we could restructure the grammar so that we have `QuantizedElementType ::= PerTensorQuantizedElementType | PerAxisQuantizedElementType`, and I gave it a try but that leads to non-trivial duplication, so overall I don't think this is worth it.

Reply:
makes sense.